src/core/inductive.ml

```diff
@@ -125,10 +191,12 @@ let fold_cons_typ (pos : popt) (codom : term list -> 'b list -> 'a -> 'c)
     ind_T: Π(p:Π(i1:A1),...,Π(im:Am), T i1 ... im -> Prop), U1 -> ... -> Un ->
     (Π(i1:A1),...,Π(im:Am), Π(t:T i1 ... im), π(p i1 ... im t))
     where Ui is the clause associated to the constructor ci. *)
-let gen_rec_type : Sig_state.t -> popt -> sym -> sym list -> term =
-  fun ss pos ind_sym cons_list ->
+let gen_rec_type :
+      Sig_state.t -> popt -> sym list -> (sym list) list
+      -> (term list * (sym * (tvar * tbox)) list) =
+  fun ss pos ind_typ_list cons_list_list ->
 
-  (* STEP 0: Define some tools which will be useful *)
+  (* STEP 1 - Do preprocessing *)
   let c = get_config ss pos in
   let set =
```

Comment:
Some comment is missing here. What is `set` for? And `set` is too generic. You should rename it into say `names`.

Reply:
set -> set_names_clash